docs/src/main/asciidoc/_migration-guide-for-4.0.adoc

```diff
@@ -1395,7 +1417,16 @@ spring:
               entity-type: queue #set as topic if needed
 ----
 
-To use the security principal to retrieve a connection string for authentication, taking service principal as example, configuration changes are listed as below:
+If you use other security principals instead of connection strings, in the legacy versions the application will firstly connect to Azure Resource Manager (ARM) with the provided security principal, and then
+retrieve the connection string of the specified namespace with ARM. In the end the application uses the retrieved connection string to authenticate to an Azure Service. In this way
+the provided security principal should be granted with the link:https://docs.microsoft.com/azure/role-based-access-control/built-in-roles#contributor[Contributor] role to retrieve of the associated Azure Service namespace.
+
+For Azure Spring Cloud 4.0, we provide two ways of leveraging security principals for authentication. One is the legacy usage, the principals are used to
+connect to ARM and then retrieve connection string as credentials, where the `Contributor` role is required.
```

Comment:
remove the `as credentials`, connection strings contains credential information, but they != credentials